### PR TITLE
Change the default ssh config path to the home directory of pipecd user

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipe/issues/987

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/git/ssh_config.go [hunk 1]
```diff
@@ -33,7 +33,10 @@ Host {{ .Host }}
     StrictHostKeyChecking no
 `
 
-var sshConfigTmpl = template.Must(template.New("ssh-config").Parse(sshConfigTemplate))
+var (
+	sshConfigTmpl            = template.Must(template.New("ssh-config").Parse(sshConfigTemplate))
+	defaultSSHConfigFilePath = "/home/pipecd/.ssh/config"
```
Comment (nakabonne) [inline pkg/git/ssh_config.go:38]: Alternatively, `$HOME/.ssh/config`?

Comment (nghialv) [reply pkg/git/ssh_config.go:38]: The current one is fine to be unified with other places.

Comment (nakabonne) [reply pkg/git/ssh_config.go:38]: 👍

**Pull request merged**